interpreter.js

```diff
@@ -1921,6 +1921,24 @@ Interpreter.prototype.pseudoToNative = function(pseudoObj, opt_cycles) {
   return nativeObj;
 };
 
+/**
+ * Constructs the fully qualified name of a child node. i.e. baz->foo.bar.baz.
+ * @param {node} object The child node of the object call stack
+ * @return {string} The string representation of the object call stack.
+ */
+Interpreter.prototype.getFullyQualifiedName = function(node) {
+  var objectCallStack = [];
+  while (node.object && node.property) {
+    objectCallStack.push(node.property.name);
+    node = node.object;
+  }
+  var fullyQualifiedName = node.name;
+  while (0 < objectCallStack.length) {
+    fullyQualifiedName += '.' + objectCallStack.pop();
```

Comment:
is there a specific reason to manually assemble this rather than just using array.join?